catalog/core/catalog-core-commands/src/test/java/org/codice/ddf/commands/catalog/SeedCommandTest.java

```diff
@@ -285,4 +305,61 @@ public boolean matches(Object o) {
             return test.test((QueryRequest) o);
         }
     }
+
+    private void mockMultipleQueries(int pageSize, int resourceLimit) throws Exception {
+
+        mockResourceResponse();
+        QueryResponse queryResponseMock = mock(QueryResponse.class);
+
+        // have as many results as the resource limit will allow
+        List<Result> resultsList = populateResultMockList(resourceLimit);
+        List<Result>[] queriedResultsList = mockQueryGetResults(resultsList, pageSize);
+
+        when(queryResponseMock.getResults()).thenReturn(queriedResultsList[0], queriedResultsList);
+        when(catalogFramework.query(any())).thenReturn(queryResponseMock);
+
+        seedCommand.resourceLimit = resourceLimit;
+    }
+
+    private List<Result> populateResultMockList(int size) {
+
+        List<Result> resultMockList = new ArrayList<>();
+
+        for (int i = 0; i < size; i++) {
+            resultMock = mock(Result.class);
+            metacardMock = mock(Metacard.class);
+
+            when(resultMock.getMetacard()).thenReturn(metacardMock);
+            when(metacardMock.getId()).thenReturn("MOCK METACARD " + (i + 1));
+
+            resultMockList.add(resultMock);
+        }
+        return resultMockList;
+    }
+
+    /**
+     *     Divides results list by the page size. Used for thenReturn statements.
+     *     The array being passed will have doubles for every query after the first,
+     *     for example:
+     *          [ query1, query2, query2, ... ]
+     *     This is because the query result will need to be returned twice for every
+     *     iteration, and the first query is get inserted in the first parameter
+     *     of thenReturn.
+     *     i.e. thenReturn(query1, [ query1, query2, query2 ]).
+     */
+    private List<Result>[] mockQueryGetResults(List<Result> results, int pageSize) {
+
+        int queries = (int) Math.ceil(results.size() / pageSize);
+        List<List<Result>> pageResults = Lists.partition(results, pageSize);
+        List[] pageArray = new List[queries * 2 - 1];
+
+        pageArray[0] = pageResults.get(0);
+
+        for (int i = 1, j = 1; j < pageResults.size(); i += 2, j++) {
+            pageArray[i] = pageResults.get(j);
+            pageArray[i + 1] = pageResults.get(j);
+        }
```

Comment:
Though at this point I suppose its starting to get uglier and longer, so it may not be the best option. If you can find a way to succinctly express it then you may, but the final choice is up to you.